lib/readline.js

```diff
@@ -94,10 +95,16 @@ function Interface(input, output, completer, terminal) {
     if (input.prompt !== undefined) {
       prompt = input.prompt;
     }
+    if (typeof input.escapeCodeTimeout === 'number' &&
```

Comment:
Perhaps `Number.isFinite()` should be used here. It's also probably better to throw, rather than silently use a different value.